addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -420,75 +413,33 @@ function makeActionManager(env) {
         return index;
     }
 
-    /**
-     * Triggers a re-rendering with respect to the given controller.
-     *
-     * @private
-     * @param {Controller} controller
-     * @param {UpdateStackOptions} options
-     * @param {boolean} [options.clearBreadcrumbs=false]
-     * @param {number} [options.index]
-     * @returns {Promise<Number>}
-     */
-    async function _updateUI(controller, options = {}) {
-        let resolve;
-        let reject;
-        let dialogCloseResolve;
-        const currentActionProm = new Promise((_res, _rej) => {
-            resolve = _res;
-            reject = _rej;
-        });
-        const action = controller.action;
-
-        class ControllerComponent extends Component {
-            setup() {
-                this.Component = controller.Component;
-                this.componentRef = hooks.useRef("component");
-                this.registerCallback = null;
-                if (action.target !== "new") {
-                    let beforeLeaveFn;
-                    this.registerCallback = (type, fn) => {
-                        switch (type) {
-                            case "export":
-                                controller.getState = fn;
-                                break;
-                            case "beforeLeave":
-                                beforeLeaveFn = fn;
-                                break;
-                        }
-                    };
-                    useBus(env.bus, "CLEAR-UNCOMMITTED-CHANGES", (callbacks) => {
-                        if (beforeLeaveFn) {
-                            callbacks.push(beforeLeaveFn);
-                        }
-                    });
-                }
-            }
-            catchError(error) {
-                reject(error);
+    function _buildActionControllerCallbacks(options) {
+        return {
```

Comment:
[STYLE]
define each function at the topmost indentation level:
`function onMounted() {}`

and then `return { onMounted }`

Reply:
It may be interesting to separate these callbacks by role.
eg: OnMounted and catchError have a transaction control role
while onTitleUpdated and onHistroyBack have more of a flow control role